src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -94,19 +98,21 @@ public static Entity createGameEntity(String gameName, DatastoreService datastor
   * add user to user list in game entity, and creates a score entity for the given user and game
   */
   public static boolean addUserToGame(String userId, Entity gameEntity, DatastoreService datastore) {
- 
+
     if(userId == ""){
       return false;
     }
-
-    if(gameEntity == null || datastore == null){
-      log.severe("null entity or datastore"); 
-      return false;
+    if(gameEntity == null){
+        log.severe("found null game entity trying to user "+ userId +" to game");
+        return false;
+    }
+    if(datastore == null){
+        log.severe("found null datastore trying to game to user " + userId);
+        return false;
     }
  
     ArrayList<String> userIds = (ArrayList<String>) gameEntity.getProperty("userIds");
     if(userIds == null){
-      log.severe("empty user id list");
```

Comment:
why did u remove the log here?

Reply:
It's not a fatal error, it just means the list wasn't initialized, so that can be delt with after the function is called.